### PR TITLE
mgr/dashboard: update npm packages

### PR DESCRIPTION
**Pull request by tspmelo**
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

### CODE REVIEW

**Comment by rjfd** [comment]
@tspmelo what's the status? can we merge this?

**Comment by tspmelo** [comment]
jenkins retest this please

**Comment by votdev** [comment]
```
> fsevents@1.2.9 install /ceph/src/pybind/mgr/dashboard/frontend/node_modules/@angular/compiler-cli/node_modules/fsevents
> node install


> @angular/cli@8.1.2 postinstall /ceph/src/pybind/mgr/dashboard/frontend/node_modules/@angular/cli
> node ./bin/postinstall/script.js

? Would you like to share anonymous usage data with the Angular Team at Google under
Google’s Privacy Policy at https://policies.google.com/privacy? For more details and
how to change this setting, see http://angular.io/analytics. (y/N)
```
The question is still asked.

File: src/pybind/mgr/dashboard/frontend/src/app/shared/components/language-selector/language-selector.component.ts [hunk 1]
```diff
@@ -39,8 +39,16 @@ export class LanguageSelectorComponent implements OnInit {
     }
   }
 
+  /**
+   * Jest is being more restricted regarding spying on the reload method.
+   * This will allow us to spyOn this method instead.
+   */
+  windowReload() {
```
Comment (votdev) [inline src/pybind/mgr/dashboard/frontend/src/app/shared/components/language-selector/language-selector.component.ts]: Would it make sense to rename it to ``reloadWindow``? That seems to me to be more linguistically correct.

Comment (tspmelo) [reply src/pybind/mgr/dashboard/frontend/src/app/shared/components/language-selector/language-selector.component.ts]: Done

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
jenkins test make check

**Comment by tspmelo** [comment]
@votdev currently the only way to disable the question is exporting `NG_CLI_ANALYTICS="false"`.
I have implemented that in some of the scripts in ceph repo, but if you run `npm ci` outside of a script it will still ask you.

File: src/pybind/mgr/dashboard/frontend/src/app/app-routing.module.ts [hunk 2]
```diff
@@ -179,15 +179,15 @@ const routes: Routes = [
     canActivate: [AuthGuardService],
     canActivateChild: [AuthGuardService],
     data: { breadcrumbs: 'Pools' },
-    loadChildren: './ceph/pool/pool.module#RoutedPoolModule'
+    loadChildren: () => import('./ceph/pool/pool.module').then((m) => m.RoutedPoolModule)
```
Comment (epuertat) [inline src/pybind/mgr/dashboard/frontend/src/app/app-routing.module.ts]: Interesting change. More verbose, but if it [helps catch typos before build time](https://www.youtube.com/watch?v=jPXl7sCPCOA), I'm happy with that.

**Pull request merged**